libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslDiagnostics.java

```diff
@@ -406,4 +413,38 @@ private static boolean checkIssuer(X509Certificate certificate, X509Certificate
     private static boolean isSelfIssued(X509Certificate certificate) {
         return certificate.getIssuerX500Principal().equals(certificate.getSubjectX500Principal());
     }
+
+    private static String keyUsageDescription(X509Certificate certificate) {
+        return Optional.ofNullable(certificate.getKeyUsage())
+            .map(keyUsage -> "keyUsage [" + Arrays.toString(keyUsage) + "]")
+            .orElse("no keyUsage");
+    }
+
+    private static String extendedKeyUsageDescription(X509Certificate certificate) {
+        try {
+            return Optional.ofNullable(certificate.getExtendedKeyUsage())
+                .map(list -> generateExtendedKeyUsageDescription(list))
+                .orElse("no extendedKeyUsage");
+        } catch (CertificateParsingException e) {
+            return "invalid extendedKeyUsage [" + e.toString() + "]";
+        }
+    }
+
+    private static String generateExtendedKeyUsageDescription(List<String> list) {
+        return list.stream()
+            .reduce((x, y) -> x + ", " + y)
+            .map(str -> "extendedKeyUsage [" + str + "]")
+            .orElse("no extendedKeyUsage");
+    }
+
+    private static void addSessionDescription(SSLSession session, StringBuilder message) {
+        String cipherSuite = Optional.ofNullable(session).map(SSLSession::getCipherSuite).orElse("<unknown cipherSuite>");
+        String protocol = Optional.ofNullable(session).map(SSLSession::getProtocol).orElse("<unknown protocol>");
+        message.append("; the session supports the cipher suite [")
```

Comment:
For brevity we can omit articles from the diagnostic
```suggestion
        message.append("; the session supports cipher suite [")
```